### PR TITLE
Localtunnel Feature added

### PR DESCRIPTION
**Pull request by endbehavi0r**
New tunneling option Localtunnel is added.

### CODE REVIEW

**Comment by An0nUD4Y** [comment]
THANKS FOR THIS PR..
I will merge it after testing..

**Comment by An0nUD4Y** [comment]
@endbehavi0r 
Getting some npm version issues.
Please Share solution if you have any..

And also take a look at these files.. (https://wa4e.com/downloads/)
May be we can use those executable files in place of install whole repo.
But i also found that sometime these executables return error.

PLEASE SHARE YOUR THOUGHTS...

**Comment by endbehavi0r** [comment]
I'll check. Need some time.

**Comment by endbehavi0r** [comment]
Neither does the binary nor the npm installation works for Termux on Android. I prefer the binaries provided they do not require nodeJS. What do you think?

**Comment by Moltivie** [comment]
@endbehavi0r this is what I am getting when trying to use a custom URL.
May I get you fix this. Thanks.

image

File: Defs/Actions.py [hunk 1]
```diff
@@ -213,6 +215,16 @@ def runLocalxpose(port):
         system('clear')
         return runLocalxpose(port)        
 
+def runLT(port):
+    s=input(('{1}[leave blank for random]\n{0}hiddeneye(localtunnel/subdomain)> {2}').format(MAIN0,MAIN4,MAIN2))
+    try:
+        system('lt -p '+port+((' -s '+s) if s!='' else s)+' > link.url &')
```
Comment (Moltivie) [inline Defs/Actions.py]: @endbehavi0r `lt` command. may be not be present on the systems itself.

**Possible fixes steps:**
- A control for the presence of `lt` command. If not found, install it or require user to install it manually.

Comment (endbehavi0r) [reply Defs/Actions.py]: That control is already there in the Checks.py file.
You might have to increase the delay (line 222 of Actions.py) if the following code works.
`lt -p 8080`

Comment (An0nUD4Y) [reply Defs/Actions.py]: @endbehavi0r , I have already tested that binary executable file.
Following errors were occurred.
1) Do not supports Termux.
2) Do not support custom subdomain properly(Sometime shows error)
3) Shows error after creating a link for a little amount.
4) Created link gets disconnected during opening session.
5) Also looks like those binary files are not official.

Below are following issues with npm and node.js packages.
1) It tools a long time to complete installation.
2) Localtunnel works on fixed versions of npm.(versions it support 6, 9, 4)
3) We have to find some solution for installation of those versions.
4)  Also for Termux installation location of npm(localtunnel) is different from Linux.
5) And I think we should still find the way to get rid of installation process of npm.

Overall I think npm will work better than those binary files.
If somehow we can fix npm and localtunnel issues for both Linux and Termux.

Test carefully to get most of above errors..

**Comment by An0nUD4Y** [comment]
@endbehavi0r Can't merge it due to above mentioned points.
Also think about going back to npm and localtunnel packages.
For long term that will be a best option.

Do These Things..
1) Add both binary and npm options for localtunnel.
2) Also during installation of npm ask user if he wants to install npm packages or not.
3) Don't ask for npm binary downloding.
4) Add two option for Tunnelling named 
    Localtunnel (Binary Used)
    Localtunnel (npm packages used)
5) By doing this we will have less possiblity of not getting localtunnel links.

Share your thoughts..

**Comment by endbehavi0r** [comment]
Okey dokes,
I made a few more changes in Checks.py for the user to select npm or binaries.
If a problem arises, make a fresh clone of the repo.
Thanks.

**Comment by An0nUD4Y** [comment]
@endbehavi0r
I will test it..
& Will take a look if more changes required or not.

**Comment by An0nUD4Y** [comment]
@endbehavi0r , 
Please check and resolve the conflict ..

**Comment by endbehavi0r** [comment]
I will check the conflict but does my PR get a chance?

**Comment by endbehavi0r** [comment]
Screenshot from 2019-11-15 18-45-17
It does not appear to be an issue with my system as I can't reproduce this behavior on my system.

**Comment by An0nUD4Y** [comment]
@endbehavi0r
The conflict is in merging ..
As actions.py file which you cloned before 
Has been got some changes like..
I have defined a more function runNgrok.
Which in your case is not similar with master branch ...
Please compare Actions.py file to see conflicts in merging.

**Comment by endbehavi0r** [comment]
> @endbehavi0r
> The conflict is in merging ..
> As actions.py file which you cloned before
> Has been got some changes like..
> I have defined a more function runNgrok.
> Which in your case is not similar with master branch ...
> Please compare Actions.py file to see conflicts in merging.

I re-forked the repo and made changes necessary only for LocalTunnel.
That's why I had to make a new PR at #440

**Comment by Moltivie** [comment]
@endbehavi0r which PL is the one you are working on currently?

**Comment by endbehavi0r** [comment]
I was too lazy, so I re-forked the stuff and made changes. I made a new PR at #440 ready for merging. I'll close this one.

**Pull request closed without merging**